### DISCUSSION TITLE
[C#]请问是否能限制JS侧对CS下命名空间的访问范围？

### DISCUSSION THREAD

**Discussion by minim271** [Tencent/puerts#1512] [Q&A]
由于有计划支持Mod，支持任意命名空间下的反射难以控制，希望能对反射的范围作出限制。

直接更改PuerTS的C#侧源码进行过滤应该是可行的，但是有一定的侵入性，请问是否有相关的API能够实现需求？

**Comment by zombieyang** [comment]
文档上有篇关于Filter的内容，看完应该就解决了。之前已经有人试过拿来做mod，应该没问题

**Reply by minim271** [reply]
尝试了一下Filter，似乎只能控制生成了Warpper的类中的某些方法禁止调用，对于没生成的类好像并没起到效果，如果这不是PuerTS预期的效果还请指正。

另外根据文档尝试了```env.SetDefaultBindingMode(BindingMode.DontBinding)```，但好像没用。（调用后eval仍然可以通过CS来访问未生成Warpper的类）

具体需求来说是希望避免Mod可以直接调用System.IO之类的接口，最好的情况下是只能用特定namespace下的接口。

**Reply by minim271** [reply]
```env.SetDefaultBindingMode(BindingMode.DontBinding)```调用后eval仍然可以通过CS来访问未生成Warpper的类这一点在Unity工程中也复现了，应该不是项目配置的问题。

**Reply by minim271** [reply]
如果新版本中也有类似 #1289 中提到的直接禁用反射的方法还请指教，目前看已经没有相关常量定义了。

**Reply by zombieyang** [reply]
看来是有Bug，我在master上修复了，你合进去试试

**Reply by minim271** [reply]
确认修复，```env.SetDefaultBindingMode(BindingMode.DontBinding)```后未生成的方法已经无法调用了。

不过观察到ctor和properties是不受此限制的，是否考虑也支持限制？

**Reply by zombieyang** [reply]
我下周抽时间写Unittest把这些都检查下
